### PR TITLE
Remove Slurm V4 modules and add note to use and reference V4 modules and examples

### PR DESCRIPTION
**Pull request by harshthakkar01**
This PR removes V4 modules and adds a note to use old version of GHPC source code/ binary if customer wants to deploy V4 modules and refer old examples. 

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

**Comment by harshthakkar01** [comment]
Sending out for early review. 

TODO: Submit this after nfs-sql bp is updated to V6.

**Review by mr0re1 — APPROVED** [review]

File: modules/README.md
Comment (nick-stroud) [inline modules/README.md:1]: I am assuming that you did a repo wide search for module names (ex. `schedmd-slurm-on-gcp-controller`). Maybe also a search for `v4`.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -10,6 +10,11 @@ networking, storage, etc.) following Google Cloud best-practices, in a repeatabl
 manner. The HPC Toolkit is designed to be highly customizable and extensible,
 and intends to address the HPC deployment needs of a broad range of customers.
 
+> **_NOTE:_** Slurm V4 is deprecated. In case, you want to use V4 modules, please use
```
[the remark this answers is not in the document]
Comment (harshthakkar01) [reply README.md]: Done.

File: modules/README.md [hunk 2]
```diff
@@ -216,8 +210,8 @@ Pub/Sub subscription. Primarily used for [FSI - MonteCarlo Tutorial][fsi-monteca
 [schedmd-slurm-gcp-v5-controller]: ../community/modules/scheduler/schedmd-slurm-gcp-v5-controller/README.md
 [schedmd-slurm-gcp-v5-login]: ../community/modules/scheduler/schedmd-slurm-gcp-v5-login/README.md
 [schedmd-slurm-gcp-v5-hybrid]: ../community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md
-[schedmd-slurm-on-gcp-controller]: ../community/modules/scheduler/SchedMD-slurm-on-gcp-controller/README.md
-[schedmd-slurm-on-gcp-login-node]: ../community/modules/scheduler/SchedMD-slurm-on-gcp-login-node/README.md
+[schedmd-slurm-on-gcp-controller]: ../community/modules/scheduler/schedmd-slurm-gcp-v6-controller/README.md
+[schedmd-slurm-on-gcp-login-node]: ../community/modules/scheduler/schedmd-slurm-gcp-v6-login/README.md
```
[the remark this answers is not in the document]
Comment (harshthakkar01) [reply modules/README.md]: Done.

File: modules/README.md [hunk 3]
```diff
@@ -54,8 +54,6 @@ Modules that are still in development and less stable are labeled with the
   pool][htcondor-setup].
 * **[pbspro-execution]** ![community-badge] ![experimental-badge] :
   Creates execution hosts for use in a PBS Professional cluster.
-* **[SchedMD-slurm-on-gcp-partition]** ![community-badge] ![deprecated-badge] : Creates a partition
```
[the remark this answers is not in the document]
Comment (harshthakkar01) [reply modules/README.md:57]: Done.

File: modules/README.md
Comment (harshthakkar01) [reply modules/README.md:1]: Yes, thanks for the input. I found 1 extra link as well using v4 search. Updated that.

Comment (nick-stroud) [inline modules/README.md:1]: What is the disposition on the failing tests?

**Review by nick-stroud — APPROVED** [review]

File: modules/README.md
Comment (harshthakkar01) [reply modules/README.md:1]: * v6-tpu --> TPU not available.
* v5-ubuntu2004 --> known issue. fix is in progress.
* test-chrome-desktop --> wait for startup failing (not using slurm modules)
* hpc-slurm-chrome-desktop --> v5 module. not sure why it is failing. re-triggered.

Comment (harshthakkar01) [reply modules/README.md:1]: Offline discussion. These failures are not related to this PR.

**Pull request merged**